src/scenic/simulators/metadrive/simulator.py

```diff
@@ -139,6 +170,20 @@ def createObjectInSimulator(self, obj):
             metadrive_objects = self.client.engine.get_objects()
             obj.metaDriveActor = list(metadrive_objects.values())[0]
             self.defined_ego = True
+
+            # Adding sensors if available
+            if obj.sensors:
+                for name, sensor in obj.sensors.items():
+                    metadrive_sensor = self.client.engine.get_sensor(name)
```

Comment:
It looks like this part would go wrong if two cars have a sensor with the same name. You could append the index of the object to the name to make it unique (for example).